src/Symfony/Component/Validator/Tests/Constraints/UniqueValidatorTest.php

```diff
@@ -220,6 +221,64 @@ public function testExpectsValidCaseInsensitiveComparison()
 
         $this->assertNoViolation();
     }
+
+    public function testCollectionFieldsAreOptional()
+    {
+        $this->validator->validate([['value' => 5], ['id' => 1, 'value' => 6]], new Unique(fields: 'id'));
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getInvalidFieldNames
+     */
+    public function testCollectionFieldNamesMustBeString(string $type, mixed $field)
+    {
+        $this->expectException(UnexpectedTypeException::class);
+        $this->expectExceptionMessage(sprintf('Expected argument of type "string", "%s" given', $type));
+
+        $this->validator->validate([['value' => 5], ['id' => 1, 'value' => 6]], new Unique(fields: [$field]));
+    }
+
+    public function getInvalidFieldNames(): array
+    {
+        return [
+            ['stdClass', new \stdClass()],
+            ['int', 2],
+            ['bool', false],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidCollectionValues
+     */
+    public function testInvalidCollectionValues(array $value, array $fields)
+    {
+        $this->validator->validate($value, new Unique([
+            'message' => 'myMessage',
+        ], fields: $fields));
+
+        $this->buildViolation('myMessage')
+            ->setParameter('{{ value }}', 'array')
+            ->setCode(Unique::IS_NOT_UNIQUE)
+            ->assertRaised();
+    }
+
+    public function getInvalidCollectionValues(): array
+    {
+        return [
+            'unique string' => [[['lang' => 'eng', 'translation' => 'hi'], ['lang' => 'eng', 'translation' => 'hello'],
+            ], ['lang']],
```

Comment:
the CS is weird, we should find a way to make this more readable imho

Reply:
> the CS is weird, we should find a way to make this more readable imho

Do you mean the data in the example or the structure of the array?

Data are based on situations like:
"Multiple translations of the same phrase (like the title of the page) are provided for the same language"

Reply:
I mean the way the array is written - the indentation and (lack of) line breaks

Reply:
Whould this be better?
```php
    public function getInvalidCollectionValues(): array
    {
        return [
            'unique string' => [[
                ['lang' => 'eng', 'translation' => 'hi'], 
                ['lang' => 'eng', 'translation' => 'hello'],
            ], ['lang']],
            'unique floats' => [[
                ['latitude' => 51.509865, 'longitude' => -0.118092, 'poi' => 'capital'],
                ['latitude' => 52.520008, 'longitude' => 13.404954],
                ['latitude' => 51.509865, 'longitude' => -0.118092],
            ], ['latitude', 'longitude']],
            'unique int' => [[
                ['id' => 1, 'email' => 'bar@email.com'], 
                ['id' => 1, 'email' => 'foo@email.com'],
            ], ['id']],
        ];
    }

Reply:
yes :)

Reply:
Example reformatted.